### PR TITLE
Add CodeCov token

### PR DESCRIPTION
**Pull request by gdalle**
see https://discourse.julialang.org/t/psa-new-version-of-codecov-action-requires-additional-setup/109857

### CODE REVIEW

**Comment by gdalle** [comment]
Do not merge: codecov is failing for server error reasons
https://github.com/JuliaGraphs/Graphs.jl/actions/runs/7826337073/job/21352010540?pr=339

**Comment by gdalle** [comment]
ok we're good

**Review by simonschoelly — APPROVED** [review]
Looks good to me. I guess if we merge this, if the code coverage is indeed correctly reported on the bade.

**Comment by gdalle** [comment]
> on the bade.

what does this mean?

**Pull request merged**
